scripts/quests/ahtUrhgan/Rat_Race.lua

```diff
@@ -0,0 +1,212 @@
+                    else
+                        return quest:event(313)
+                    end
+                end,
+            },
+
+            ['Kyokyoroon'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        quest:getVar(player, 'Prog') == 5 and
+                        trade:hasItemQty(xi.items.BOWL_OF_NASHMAU_STEW, 1) and
+                        trade:getItemCount() == 1
+                    then
+                        return quest:progressEvent(311)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 6 then
+                        return quest:event(316)
+                    end
+                end,
+            },
+
+            ['Ququroon'] =
+            {
+                onTrade = function(player, npc, trade)
+                    if
+                        quest:getVar(player, 'Prog') == 4 and
+                        trade:hasItemQty(xi.items.AHTAPOT, 1) and
+                        trade:hasItemQty(xi.items.ISTAKOZ, 1) and
+                        trade:hasItemQty(xi.items.ISTAVRIT, 1) and
+                        trade:hasItemQty(xi.items.ISTIRIDYE, 1) and
+                        trade:hasItemQty(xi.items.MERCANBALIGI, 1) and
+                        trade:getItemCount() == 5
+                    then
+                        return quest:progressEvent(310)
+                    end
+                end,
+
+                onTrigger = function(player, npc)
+                    local questProgress = quest:getVar(player, 'Prog')
+
+                    if questProgress == 3 then
+                        return quest:progressEvent(309)
+                    elseif questProgress == 4 then
+                        return quest:event(242)
+                    elseif questProgress >= 5 then
+                        return quest:event(315)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [309] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 4)
+                end,
+
+                [310] = function(player, csid, option, npc)
+                    if npcUtil.giveItem(player, xi.items.BOWL_OF_NASHMAU_STEW) then
+                        player:tradeComplete()
+                        quest:setVar(player, 'Prog', 5)
+                    end
+                end,
+
+                [311] = function(player, csid, option, npc)
+                    player:tradeComplete()
+                    quest:setVar(player, 'Prog', 6)
+                end,
+
+                [312] = function(player, csid, option, npc)
+                    -- TODO: Modify npcUtil.giveItem to support multiple reward/unable to obtain messages.
+                    if player:getFreeSlotsCount() <= 2 then
```

Comment:
Can check later and see if retail even generates a message for all items if no slots and gaining more than 1 item. It could just fail and list the 1st item in the altercation allowing the util to work the same way